File: app/src/main/java/com/iug/coursehub/CourseHubApp.java
package com.iug.coursehub;

import android.app.Application;

import com.iug.coursehub.data.local.db.CoursesRepository;
import com.iug.coursehub.data.local.prefs.AppPrefs;

public class CourseHubApp extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        CoursesRepository.initDb(this);

        AppPrefs.initPrefs(this);
        AppPrefs.saveAdminEmail();
        AppPrefs.saveAdminPassword();
    }
}

File: app/src/main/java/com/iug/coursehub/CourseHubApp.kt
package com.iug.coursehub

import android.app.Application
import com.iug.coursehub.data.local.db.CoursesRepository
import com.iug.coursehub.data.local.prefs.AppPrefs

class CourseHubApp : Application() {
    override fun onCreate() {
        super.onCreate()
        CoursesRepository.initDb(this)
        AppPrefs.apply {
            initPrefs(this@CourseHubApp)
            saveAdminEmail()
            saveAdminPassword()
        }
    }
}